### ISSUE TITLE
`check_assertion_p` batching rule returns a tuple as the output batch dimension; `vmap` fails over `check_assertions()`

### ISSUE THREAD

**Issue by jonkhler** [cusp-ai-oss/kUPS#247]
## Summary

`_make_constant_primitive` in `src/kups/core/assertion.py` registers a batching rule for `check_assertion_p`. The primitive has `multiple_results = False`. JAX expects one output batch dimension: an `int` or `None`. The rule returns the input `batch_dims` tuple. JAX rejects the tuple.

## Impact

`jax.vmap` fails over any computation that contains `check_assertions()`.

## Environment

- kups 1.0.3 (`1b3d70d`)
- jax / jaxlib 0.11.0
- Python 3.12

## Steps to reproduce

Run this script:

```python
import jax, jax.numpy as jnp
from kups.core.assertion import check_assertions

jax.vmap(lambda x: check_assertions(x))(jnp.ones(4))
```

## Observed result

```
Exception: batch dim should be int or `None`
```

## Cause

```python
def constant_p_batcher(
    batched_args: tuple[Any, ...], batch_dims: tuple[Any, ...], **kwargs: Any
) -> tuple[Any, tuple[Any, ...]]:
    return primitive.bind(*batched_args, **kwargs), batch_dims
```

## Possible fix

```diff
-        return primitive.bind(*batched_args, **kwargs), batch_dims
+        return primitive.bind(*batched_args, **kwargs), None
```

The output is a scalar. Its value does not depend on the batch axis. Thus `None` is the correct batch dimension. An alternative design returns `0` and binds one scalar for each batch element. That design changes the output semantics. Keep `None` to match the current semantics.

## Verification

Apply the fix. Run the script again. The call returns a scalar `True` result, broadcast by vmap to shape `(4,)`.
